src/Slider/Slideshow.js

```diff
@@ -35,17 +38,35 @@ const SlideshowDot = styled.div`
   background-color: ${(props) => (props.active ? "#303030" : "#c4c4c4")};
 `;
 
+const SlideText = styled.div`
+  position: absolute;
+  bottom: 10px;
+  right: 10px;
+  text-align: right;
+  color: white;
+  text-shadow: 2px 2px 2px #000000;
+  font-size: 1.2em;
+`;
+
 function Slideshow({ slides }) {
   const [index, setIndex] = React.useState(0);
 
   return (
     <SlideshowContainer>
       <SlideshowSlider index={index}>
-        {slides.map((slide, index) => (
-          <div key={index}>
-            <img src={slide.url} alt={slide.alt} />
-          </div>
-        ))}
+        {slides.map((slide, index) =>
+          slide.navigateTo === undefined ? (
+            <div key={index}>
+              <img src={slide.url} alt={slide.alt} />
+              <SlideText>{slide.text.toUpperCase()}</SlideText>
+            </div>
+          ) : (
+            <Link as="div" to={slide.navigateTo} key={index}>
+              <img src={slide.url} alt={slide.alt} />
+              <SlideText>{slide.text.toUpperCase()}</SlideText>
+            </Link>
```

Comment:
Creo que esta repetición de código se puede solucionar con un High Order Component, que reciba sus children como prop y que solo se encargue de ver si el wrapper de lo mismos va a ser un `div` normal o un `Link`, esa lógica se la dejaríamos al HOC para en este componente hacer algo como: 
```suggestion
            <Slide as="div" to={slide.navigateTo} key={index}>
              <img src={slide.url} alt={slide.alt} />
              <SlideText>{slide.text.toUpperCase()}</SlideText>
            </Slide>
```

Evitando que hacer la validación de `slide.navigateTo` aquí y reduciendo unas 10 líneas de código a solo 4.

Hazme saber si necesitas ayuda para concebir ese HOC, podríamos agendar algo para explicar más a fondo esta implementación. E incluso programarla juntos.

Reply:
Ah! Esto esta muy interesante, implemente una version de lo que dices, igual y la puedes revisar en mi nuevo deliverable a ver si si es a lo que te referias